pkg/resource/replication_group/delta_util.go

```diff
@@ -62,3 +80,19 @@ func engineVersionsMatch(
 
 	return false
 }
+
+// logDeliveryRequiresUpdate retrieves the last requested configurations saved in annotations and compares them
+// to the current desired configurations
+func logDeliveryRequiresUpdate(desired *resource) bool {
+	var desiredConfigs []*svcapitypes.LogDeliveryConfigurationRequest
+	var lastRequestedConfigs []*svcapitypes.LogDeliveryConfigurationRequest
+
+	desiredConfigs = desired.ko.Spec.LogDeliveryConfigurations
+
+	annotations := desired.ko.ObjectMeta.GetAnnotations()
+	if val, ok := annotations[AnnotationLastRequestedLDCs]; ok {
+		_ = json.Unmarshal([]byte(val), &lastRequestedConfigs)
+	}
```

Comment:
error can ignored here for a reason?

Reply:
During marshaling, if there is an error, we set the value to the string literal "null" (as discussed above).

Whatever value we unmarshal from `annotations[AnnotationLastRequestedLDCs]` will either be "null" or some valid string representation of the `Spec.LogDeliveryConfigurations` field.

So it's hard to imagine a situation in which `Unmarshal` would result in a error. Even if it does and `lastRequestedConfigs` is populated incorrectly/incompletely, this would just result in adding a delta for `LogDeliveryConfigurations` and the controller would attempt another modification with the desired state, which I think is an okay outcome.

I considered adding logging here in the case of an error, but that would involve adding the resourceManager as a receiver on this function and consequently `filterDelta` and `newResourceDelta`, which may involve a change to the template for `delta.go` (unless there's an easier way to add logging that I'm not aware of).

Another possible solution is just to `panic()` here to convey the error, but that seems a little heavy-handed to me. 

I think the status quo is okay but let me know what you think.